src/UNIV2LPOracle.sol

```diff
@@ -264,14 +294,67 @@ contract UNIV2LPOracle {
         );
     }
 
-    function poke() external stoppable {
-        require(pass(), "UNIV2LPOracle/not-passed");
-        uint128 val = seek();
-        require(val != 0, "UNIV2LPOracle/invalid-price");
-        cur = nxt;
-        nxt = Feed(val, 1);
-        zzz = uint64(block.timestamp);
-        emit Value(cur.val, nxt.val);
+    function poke() external {
+
+        // Ensure a single SLOAD while avoiding solc's excessive bitmasking bureaucracy.
+        uint256 _zph;
+        uint256 _hop;
+        {
+            uint256 _stopped;  // block-scoping _stopped here saves a little gas
+            assembly {
+                let _slot1 := sload(1)
+                _stopped   := and(_slot1,         0xff    )
+                _hop       := and(shr(8, _slot1), 0xffffff)
+                _zph       := shr(32, _slot1)
+            }
+
+            // When stopped, values are set to zero and should remain such; thus, disallow updating in that case.
+            require(_stopped == 0, "UNIV2LPOracle/is-stopped");
+        }
+
+        // Equivalent to requiring that pass() returns true;
+        // code has been duplicated to reduce gas costs.
+        require(block.timestamp >= _zph, "UNIV2LPOracle/not-passed");
```

Comment:
Did you forget to copy/paste the Yul optimizations you did in `pass()`?

Reply:
Nevermind, you did load `_zph` in the same way

Reply:
No. There'd be no reason to copy that code here as the structure is different.

Probably just the comment is confusing, since the code used to be duplicated, but now is not. I'll make it clearer.

Reply:
https://github.com/makerdao/univ2-lp-oracle/pull/41/commits/b65cf2f5ca8c07a74ac166f0844a0cf42bd6d246